sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/SnowflakeIO.java

```diff
@@ -457,22 +474,22 @@ private String makeTmpDirName() {
 
     private static class CopyIntoStageFn extends DoFn<Object, String> {
       private final SerializableFunction<Void, DataSource> dataSourceProviderFn;
-      private final String query;
-      private final String table;
-      private final String database;
-      private final String schema;
+      private final ValueProvider<String> query;
+      private final ValueProvider<String> database;
+      private final ValueProvider<String> schema;
+      private final ValueProvider<String> table;
+      private final ValueProvider<String> storageIntegrationName;
+      private final ValueProvider<String> stagingBucketDir;
       private final String tmpDirName;
-      private final String storageIntegrationName;
-      private final String stagingBucketDir;
       private final SnowflakeService snowflakeService;
       private final String quotationMark;
```

Comment:
Can we pass `quotationMark` as option ? Maybe is worth to make this var supported at runtime ? We can make Issue for this if currently there is no time for this kind of improvements.

Reply:
Sure

Reply:
And done